src/routes/signin/providers/utils.ts

```diff
@@ -184,6 +190,14 @@ const providerCallback = asyncWrapper(
 
     const user = req.user as UserFieldsFragment;
 
+    if (!user.id) {
+      const failureRedirect = new URL(requestOptions.redirectTo);
+      failureRedirect.searchParams.set('loginFailed', 'true');
+      return sendError(res, 'provider-login-failed', {
+        redirectTo: failureRedirect.toString(),
```

Comment:
we are using `signin` instead of `login`. Could you rename to `provider-signin-failed`?

Reply:
If there is a means to know sign in failed because Oauth sign-up is disabled, that would be great to adapt the error to provide more information to the frontend about why it failed